src/main/scala/io/druid/indexer/spark/SparkBatchIndexTask.scala

```diff
@@ -259,6 +261,8 @@ object SparkBatchIndexTask
   private val DEFAULT_TARGET_PARTITION_SIZE: Long   = 5000000L
   private val CHILD_PROPERTY_PREFIX        : String = "druid.indexer.fork.property."
   val log            = new Logger(SparkBatchIndexTask.getClass)
+  val lifecycle      = SerializedJsonStatic.lifecycle
```

Comment:
I moved `lifecycle.stop()` to the `SparkListener` so it will close the life cycle when spark context gets stopped. `ShutdownHookManager` seems to be an internal class so I don't think I can use it unfortunately.